### DISCUSSION TITLE
Merge polygons using Nonzero Winding Number Rule

### DISCUSSION THREAD

**Discussion by Rodrigodd** [shapely/shapely#2167] [Q&A]
I am working in processing PDF files. In PDFs, paths are commonly draw using the Nonzero Winding Number Rule (section 8.5.3.3.2 of the PDF 1.7 spec). Is there any easy way of converting the paths in the PDF to a valid shapely `Polygon`/`MultiPolygon`?

**Comment by mwtoews** [comment]
Unfortunately, I don't believe there is any specific functionality to evaluate nonzero winding numbers. Self-crossing lines are non-simple, and invalid for (e.g.) `POLYGON ((0 0, 9 9, 9 5, 0 5, 0 0))`. These are better represented as MultiPolygon objects (e.g.) `MULTIPOLYGON (((0 0, 0 5, 5 5, 0 0)), ((5 5, 9 9, 9 5, 5 5)))`, which can be done with [`shapely.make_valid(geometry)`](https://shapely.readthedocs.io/en/stable/reference/shapely.make_valid.html).
![image](https://github.com/user-attachments/assets/c812ddb0-1605-4add-bf3a-1681f43a2bfc)

**Reply by Rodrigodd** [reply]
Yeah, I am already using `make_valid` for even-odd paths, [as the documentation here](https://shapely.readthedocs.io/en/latest/reference/shapely.make_valid.html) says it uses an even-odd strategy with the linework method, although I didn't check how good it  handles it. It probably does not try very hard to handle self-intersecting paths?

But I am at least interest in unifying multiple non-intersecting subpaths (which subpath is converted to a `Polygon`). For the even-odd rule, I am using [`simmetric_difference_all`](https://shapely.readthedocs.io/en/latest/reference/shapely.symmetric_difference_all.html#shapely.symmetric_difference_all) (also didn't check if it is doing what I am expecting), but could not find anything similar for the non-zero Winding rule.

A method that could split two shapes into overlapping and non-overlapping shapes (and a n-ary variation of that) could help with that, I only would need to separately compute the winding-number for each region. But I didn't find such method, and implementing that in Python may not be worth it.
